libevmasm/Assembly.cpp

```diff
@@ -1457,6 +1482,25 @@ LinkerObject const& Assembly::assembleEOF() const
 				appendBigEndianUint16(ret.bytecode, item.data());
 				break;
 			}
+			case CallF:
+			{
+				ret.bytecode.push_back(static_cast<uint8_t>(Instruction::CALLF));
+				solAssert(item.data() <= std::numeric_limits<uint16_t>::max(), "Invalid callf index value.");
+				appendBigEndianUint16(ret.bytecode, item.data());
+				break;
+			}
+			case JumpF:
+			{
+				ret.bytecode.push_back(static_cast<uint8_t>(Instruction::JUMPF));
+				solAssert(item.data() <= std::numeric_limits<uint16_t>::max(), "Invalid jumpf index value.");
+				appendBigEndianUint16(ret.bytecode, item.data());
+				break;
+			}
+			case RetF:
+			{
+				ret.bytecode.push_back(static_cast<uint8_t>(Instruction::RETF));
+				break;
+			}
```

Comment:
This should have sanity checks that the numbers of inputs and outputs actually match the function. It's technically possible to create `AssemblyItem` without using the static helpers and get it wrong (or we might just forget when implementing asm import later). We should be able to catch this kind of mistake.